uwblueprint/paramedics-react
# Map tap functionality

[Notion ticket](https://www.notion.so/uwblueprintexecs/396370c5e09f4c28bc7f455a7e84c908?v=3d4d369c668e4d2581a2517666768b44&p=35bf1c9c71c74ca397d6240546e79b31)

## Changes
- Once a user taps on a particular location on the map, the 'Add Pin' sidebar will open with the corresponding address
- If a user adds a pin using the 'Add Pin' sidebar, then they have the option to tap on the map to designate the location for the pin. If they select a suggestion, a pin should appear at the location
- Tapping is now integrated to edit a pin's location (i.e. if you want to edit a pin's location, you can go through the same flow as listed above and simply tap the map to change the location)
- Marker should not shift anymore as you move around the map
- Integrated geocoding to convert latitude and longitude to addresses

## Screenshots
![Kapture 2020-12-08 at 13 08 16](https://user-images.githubusercontent.com/21224282/101535858-76eac480-3956-11eb-9f35-6fd3bac267d6.gif)

## Testing
**Please use ngrok to test on an iPad!!**
- Add a pin via button
- Add a pin via tap
- Add a pin via button and change address. The suggestion you click should be what is filled in the 'Pin Location' field
- Do above but use tap
- Edit a pin and change the location using tap
- Edit a pin and change the location using the autocompletion field
- Move around map and make sure markers do not move (esp. zooming in and out)

## Checklist
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

before opening PR

filling out PR

after opening PR
